### ISSUE TITLE
Redirect not working

### ISSUE THREAD

**Issue by bert-bruggeman** [bashlk/staticman-netlify-function#5]
Redirect doesn't seem to work on my new project (worked fine on previous projects)

**form.html** (https://fb-bouwteam.netlify.app/contact): 
```
<form method="post" action="https://fb-bouwteam-staticman.netlify.app/.netlify/functions/staticman/fbbouwteam/submissions/master/submissions">
<input type="hidden" name="fields[form]" value="Contactformulier">
<input name="options[redirect]" type="hidden" value="https://fb-bouwteam.netlify.app/contact/#success">
```
**staticman.yml** (the same as previous projects)
```
# Name of the property. You can have multiple properties with completely
# different config blocks for different sections of your site.
# For example, you can have one property to handle comment submission and
# another one to handle posts.
submissions:
  # (*) REQUIRED
  #
  # Names of the fields the form is allowed to submit. If a field that is
  # not here is part of the request, an error will be thrown.
  allowedFields: ['form', 'name', 'email', 'message', 'custom']

  # (*) REQUIRED
  #
  # Name of the branch being used. Must match the one sent in the URL of the
  # request.
  branch: "master"

  # Text to use as the commit message or pull request title. Accepts placeholders.
  commitMessage: "Added submission (Staticman)" # [skip ci]

  # (*) REQUIRED
  #
  # Destination path (filename) for the data files. Accepts placeholders.
  filename: "{@timestamp}-{fields.name_slug}"

  # The format of the generated data files. Accepted values are "json", "yaml"
  # or "frontmatter"
  format: "frontmatter"
  extention: "md"

  # List of fields to be populated automatically by Staticman and included in
  # the data file. Keys are the name of the field. The value can be an object
  # with a `type` property, which configures the generated field, or any value
  # to be used directly (e.g. a string, number or array)
  generatedFields:
    name_slug:
      type: slugify
      options:
        field: name
    date:
      type: date

  # Whether entries need to be appproved before they are published to the main
  # branch. If set to `true`, a pull request will be created for your approval.
  # Otherwise, entries will be published to the main branch automatically.
  moderation: false

  # Name of the site. Used in notification emails.
  name: "Website"

  # Notification settings. When enabled, users can choose to receive notifications
  # via email when someone adds a reply or a new comment. This requires an account
  # with Mailgun, which you can get for free at http://mailgun.com.
  #notifications:
    # Enable notifications
    #enabled: true

    # (!) ENCRYPTED
    #
    # Mailgun API key
    #apiKey: "1q2w3e4r"

    # (!) ENCRYPTED
    #
    # Mailgun domain (encrypted)
    #domain: "4r3e2w1q"

  # (*) REQUIRED
  #
  # Destination path (directory) for the data files. Accepts placeholders.
  path: "submissions/{options.slug}"

  # Names of required fields. If any of these isn't in the request or is empty,
  # an error will be thrown.
  requiredFields: ["email", "name"]

  # List of transformations to apply to any of the fields supplied. Keys are
  # the name of the field and values are possible transformation types.
  transforms:
    message: frontmatterContent

```

**Comment by bashlk** [comment]
Hi @bert-bruggeman 

Can you post the log from netlify functions after doing a redirect? You can find it within netlify admin like so
![image](https://user-images.githubusercontent.com/11556301/93503460-63602c80-f918-11ea-9489-5ff4d444f897.png)

When you say it worked fine in other projects, do you mean other projects also using staticman-netlify-function or just plain staticman? If it was also with staticman-netlify-function, do you notice any difference in the form implementation in HTML? (The redirect is not affected by the configuration - it is completely determined by the HTML form)

**Comment by bert-bruggeman** [comment]
Thanks for the quick response!

- I'm using your staticman (netlify) function on all my websites since I found your repo (and love it!)
- I'm using the same (hugo) form partial as in other projects, with only changes to some custom fields
- Everything else is working...


**functions log:** 
```

5:24:52 PM: 2020-09-17T17:24:52.258Z	c5941ee6-e668-411e-9a8a-cc050393400c	WARN	Deprecation: [@octokit/rest] octokit.authenticate() is deprecated. Use "auth" constructor option instead.
    at authenticate (/var/task/src/node_modules/@octokit/rest/plugins/authentication-deprecated/authenticate.js:11:5)
    at new GitHub (/var/task/src/node_modules/staticman/lib/GitHub.js:39:16)
    at Object.module.exports.create (/var/task/src/node_modules/staticman/lib/GitServiceFactory.js:11:14)
    at new Staticman (/var/task/src/node_modules/staticman/lib/Staticman.js:32:25)
    at Runtime.exports.handler (/var/task/src/staticman.js:15:21)
    at Runtime.handleOnce (/var/runtime/Runtime.js:66:25)
5:24:52 PM: 2020-09-17T17:24:52.460Z	c5941ee6-e668-411e-9a8a-cc050393400c	INFO	Warning: configuration param 'extention' not declared in the schema
5:24:52 PM: 2020-09-17T17:24:52.475Z	c5941ee6-e668-411e-9a8a-cc050393400c	WARN	Deprecation: [@octokit/rest] octokit.repos.createFile() has been renamed to octokit.repos.createOrUpdateFile() (2019-06-07)
    at Object.deprecatedEndpointMethod [as createFile] (/var/task/src/node_modules/@octokit/rest/plugins/register-endpoints/register-endpoints.js:56:13)
    at /var/task/src/node_modules/staticman/lib/GitHub.js:96:58
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
5:24:53 PM: Duration: 1179.70 ms	Memory Usage: 149 MB	Init Duration: 1319.42 ms	
```

**Comment by bashlk** [comment]
OK so I took a look at it and at first, I was a bit confused because I understood your message to mean that the redirects worked on your previous projects and I soon realised, they couldn't never have worked due to the way I hook into the staticman core 😅. 

Actually, I didn't know that this option existed because it is only mentioned briefly in the get started guide (and I skipped that bit 😅) and these custom options don't seem to be explicitly documented anywhere - raised here - https://github.com/eduardoboucas/staticman.net/issues/12.

TLDR; I wrote a quick patch that should fix the redirect. Give it a try and let me know whether it works 😄

**Comment by bert-bruggeman** [comment]
I'm feeling stupid... forgot I wrote a small inline js a long time ago to handle the redirect.
Thanks for the patch, will try it this evening... sorry for the confusion!

**Comment by bert-bruggeman** [comment]
Tried the updated version, the redirect works after solving a syntax error on line 38 removed double "})"

Thanks!

here's the updated script:

```
const qs = require('qs');
const UrlPattern = require('url-pattern');
const Staticman = require('staticman/lib/Staticman');

const pattern = new UrlPattern('/.netlify/functions/staticman/:username/:repository/:branch/:property(/)');

exports.handler = (event, context, callback) => {
  const urlParams = pattern.match(event.path);
  const params = {
    ...urlParams,
    // staticman lib has different logic per version. Lock it to 2
    version: '2',
  };

  const staticman = new Staticman(params);
  staticman.setConfigPath();
  staticman.setIp(event.headers['client-ip']);
  staticman.setUserAgent(event.headers['user-agent']);

  const body = qs.parse(event.body);
  const options = body.options || {};

  staticman.processEntry(body.fields, options).then(() => {
    if (options.redirect) {
      callback(null, {
        statusCode: 303,
        headers: {
          Location: options.redirect
        }
      });
    } else {
      callback(null, {
        statusCode: 200,
        body: 'Success',
      });
    }
  }).catch ((error) => {
  console.log(error);
  if (options.redirectError) {
    callback(null, {
      statusCode: 303,
      headers: {
        Location: options.redirectError
      }
    });
  } else {
    callback(error, {
      statusCode: 500,
      body: JSON.stringify(error)
    });
  }
});
};
```

**Comment by bashlk** [comment]
Whoops, that's what I get for being too lazy to setup a linter 😅 - thanks for pointing it out.

Will close the issue now, I hope you find this function useful for future projects as well 😄
